modules/standard/Lists.chpl

```diff
@@ -0,0 +1,364 @@
+      :rtype: `int`
+    */
+    proc count(x: this._etype): int {
+      return 0;
+    }
+
+    /*
+      Sort the items of this List in place. The parameters of this method are
+      holdovers from Python. This method could more closely integrate with
+      Chapel sorting APIs.
+
+      :arg key: Unused in current API.
+      :arg reverse: True if this List should be sorted in reverse order.
+
+    */
+    proc sort(key: this._etype=nil, reverse: bool=false) {}
+
+    /*
+      Reverse the elements of this List in place.
+    */
+    proc reverse() {}
+
+    /*
+      Index this List via subscript. Returns a reference to the element at a
+      given index in this List.
+
+      :arg i: The index of the element to access.
+
+      :return: An element from this List.
+
+      :throws IllegalArgumentError: If the given index is out of bounds.
+    */
+    proc this(i: int) ref throws {
+      return nil;
+    }
+
+    /*
+      Produce a serial iterator over the elements of this List.
+
+      :yields: A reference to one of the elements contained in this List.
+    */
+    iter these() ref {
+      yield _size;
+    }
+
+    /*
+      Write the contents of this List to a given channel.
+
+      :arg ch: A channel to write to.
+    */
+    proc writeThis(ch: channel) {}
+
+    /*
+      Convert the contents of this List to string form. Could be replaced by
+      an overload of the cast operation?
+
+      :return: The contents of this List, in a form suitable for printing.
+      :rtype: `string`
+    */
+    proc toString(): string {
+      return "";
+    }
+
+    /*
+      Return the number of elements in this List.
+
+      :return: The number of elements in this List.
+      :rtype: `int`
+    */
+    proc size {
+      return _size;
+    }
+
+    /*
+      Returns `true` if this List contains zero elements.
+
+      :return: `true` if this List is empty.
+      :rtype: `bool`
+    */
+    proc isEmpty(): bool {
+      return true;
+    }
+
+    /*
+      Returns a new DefaultRectangular array containing the elements of this
+      List.
+
+      :return: A new DefaultRectangular array.
+    */
+    proc toArray(): [] this._etype {
+      return nil;
+    }
+  }
+
+  proc =(a: List(?t), b: List(t)) {}
+
+  /*
+    Returns `true` if the contents of two Lists are the same.
+
+    :arg a: A List to compare.
+    :arg b: A List to compare.
+
+    :return: `true` if the contents of two Lists are the same.
+    :rtype: `bool`
+  */
+  proc ==(a: List(?t), b: List(t)): bool {
+    return true;
+  }
+
+  /*
+    An important question here: do we want the semantics of addition to mirror
+    Chapel arrays, or Python lists?
+
+    Given Chapel semantics, `+` would be a "vectorlike" operation, performing
+    scalar addition against all elements of this List, returning a new List
+    as a result.
+
+    Given Python semantics, `+` would be a shorthand for concatenating two
+    Lists together, returning a new List as a result.
```

Comment:
I think there should be a way to concatenate Lists but I'm not certain that it should be called `+`. I don't think `+` should promote over the list elements right now. I think we should punt on these operators for the moment and leave them out of the API, and then later decide the answer to the question above.